### PR TITLE
Add zones config UI

### PR DESCRIPTION
**Pull request by bramkragten**
Needs https://github.com/home-assistant/home-assistant/pull/30774

image
image
^ I did change that passive toggle text before commiting ;-)

### CODE REVIEW

**Comment by SeanPM5** [comment]
Looks awesome! 🎉 

One idea / suggestion to consider later on, I think radius would probably benefit from being a slider?

Feels easier to just drag a slider left or right and watch as the radius on the map grows or shrinks, rather than the tedious trial-and-error of constantly tweaking values in a text field. Especially for mobile devices where the keyboard will take up half your screen.

**Comment by bramkragten** [comment]
It should just be draggable like it is in the main view, that's something I have to add still.

File: src/components/map/ha-location-editor.ts [hunk 1]
```diff
@@ -53,11 +62,17 @@ class LocationEditor extends LitElement {
...
+    if (changedProps.has("location")) {
+      this._updateMarker();
+      this._updateRadius();
+      if (!this._ignoreFitToMap || this._ignoreFitToMap !== this.location) {
+        this.fitMap();
+      }
+      this._ignoreFitToMap = undefined;
+    }
+    if (changedProps.has("radius")) {
+      this._updateRadius();
```
Comment (balloob) [inline src/components/map/ha-location-editor.ts]: we call `updateRadius` twice.

File: src/panels/config/zone/dialog-zone-detail.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+            ? html`
+                <div class="error">${this._error}</div>
+              `
+            : ""}
+          <div class="form">
+            <paper-input
+              .value=${this._name}
+              .configValue=${"name"}
+              @value-changed=${this._valueChanged}
+              label="${this.hass!.localize("ui.panel.config.zone.detail.name")}"
```
Comment (balloob) [inline src/panels/config/zone/dialog-zone-detail.ts]:
```suggestion
              .label="${this.hass!.localize("ui.panel.config.zone.detail.name")}"
```

File: src/panels/config/zone/dialog-zone-detail.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+              error-message="${this.hass!.localize(
```
Comment (balloob) [inline src/panels/config/zone/dialog-zone-detail.ts]:
```suggestion
              .errorMessage="${this.hass!.localize(
```

File: src/panels/config/zone/dialog-zone-detail.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+                "ui.panel.config.zone.detail.required_error_msg"
+              )}"
+              .invalid=${this._name.trim() === ""}
+            ></paper-input>
+            <paper-input
+              .value=${this._icon}
+              .configValue=${"icon"}
+              @value-changed=${this._valueChanged}
+              label="${this.hass!.localize("ui.panel.config.zone.detail.icon")}"
```
Comment (balloob) [inline src/panels/config/zone/dialog-zone-detail.ts]: Same

**Review by balloob — APPROVED** [review]
🙇 

Amazing.

**Pull request merged**